### PR TITLE
Webview commands in actions

### PR DESCRIPTION
**Pull request by jsantell**
This moves the webview interactions into function action creators and a way to mock webview interactions

### CODE REVIEW

File: app/ui/browser/actions/main-actions.js [hunk 1]
```diff
@@ -77,18 +99,54 @@ export function unbookmark(url) {
...
-  return { type: types.NAVIGATE_PAGE_REFRESH, pageId };
+// @TODO Linting https://github.com/eslint/eslint/issues/3587
+export function navigatePageRefresh(pageId, doc=document) { // eslint-disable-line
+  return tagErrors((dispatch, getState) => {
+    assert(isUUID(pageId), `Invalid page id: ${pageId}`);
+    const pageIndex = getPageIndexById(getState(), pageId);
+    assert(pageIndex >= 0, `Page ${pageId} not found in current state.`);
+    assert(getState().browserWindow.pages.get(pageIndex).canRefresh, 'Page cannot refresh.');
+
+    dispatch({ type: types.NAVIGATE_PAGE_REFRESH, pageId });
```
Comment (linclark) [inline app/ui/browser/actions/main-actions.js]: Since this action type is no longer handled by the reducer, is it still an action for the sake of logging?

Comment (jsantell) [inline app/ui/browser/actions/main-actions.js]: Correct; but not sure if that's the right way to go about it -- in that case, maybe it makes sense for the reducer to fire the side effect (webview navigation) for replayability?

Comment (linclark) [inline app/ui/browser/actions/main-actions.js]: Redux, and a lot of the ecosystem around Redux, assume that the reducer has no side effects. From the Redux docs:

> It’s very important that the reducer stays pure. Things you should never do inside a reducer:
>
>    Mutate its arguments;
>    Perform side effects like API calls and routing transitions;
>    Call non-pure functions, e.g. Date.now() or Math.random().

I don't think we'll be able to achieve replayability for these actions. I might be seeing it wrong, so if you think we can maybe we should walk through a use case.

Comment (jsantell) [inline app/ui/browser/actions/main-actions.js]: Good point -- not sure about this then. Either way, can move it elsewhere later for the replayability effect if it's even possible. Does it look good?

Comment (linclark) [inline app/ui/browser/actions/main-actions.js]: In this setup, we'd send actions to the store that don't affect the store. That feels a little off to me. It means we're doing extra work (going through the reducers, triggering the rerender process) just to get a log of something.

If you feel strongly that this is the approach we should take, I won't block it. But I think we should consider removing these actions and just handling webview commands with regular click handlers, and handle logging that some other way. I think that would reduce complexity.

File: test/ui/browser/utils.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  }).join(' ');
+
+  return new Promise(resolve => {
+    const inc = (el, attr) => () => el.setAttribute(attr, ~~el.getAttribute(attr) + 1);
+    jsdom.env({
+      html: domText,
+      done(err, window) {
+        const webviews = window.document.querySelectorAll('webview');
+        Array.prototype.forEach.call(webviews, webview => {
+          webview.goBack = inc(webview, 'back-count');
```
Comment (linclark) [inline test/ui/browser/utils.js]: Would it make sense for us to use sinon test spies here? http://sinonjs.org/docs/#spies

**Comment by jsantell** [comment]
Superceded via #358

**Pull request closed without merging**